addons/payment_asiapay/views/payment_views.xml

```diff
@@ -0,0 +1,18 @@
+<?xml version='1.0' encoding='utf-8' ?>
+<odoo>
+    <record id="payment_acquirer_form" model="ir.ui.view">
+        <field name="name">Asiapay Acquirer Form</field>
+        <field name="model">payment.acquirer</field>
+        <field name="inherit_id" ref="payment.payment_acquirer_form" />
+        <field name="arch" type="xml">
+            <xpath expr='//group[@name="acquirer"]' position='inside'>
+                <group attrs="{'invisible': [('provider', '!=', 'asiapay')]}">
+                    <field name="asiapay_merchant_id" password="True" />
+                    <field name="asiapay_currency_id" password="True" />
+                    <field name="asiapay_secure_hash" password="True" />
+                    <field name="asiapay_secure_hash_function" />
```

Comment:
The fields should be required when the acquirer is not disabled (see other modules' views).

Also, we generally only set the password widget on secret fields; see [task 2694139](https://www.odoo.com/web#id=2694139&menu_id=5195&cids=1&action=333&active_id=4106&model=project.task&view_type=form). The same goes for the `groups='base.group_system'` attribute that we set on secret field for an extra layer of security.

Reply:
Just one question, the secure hash is not mandatory in Asiapay side. Although this is not safe at all, it is possible to make payment without secure hash.

Reply:
They don't make it mandatory because they want developers to integrate with them as painlessly as possible. We, however, want to make it mandatory for our users because we don't want to deal with security loopholes in their codebase. Without the security check, one can simply craft an URL in their browser to confirm their own order on an Odoo shop, for example.